src/main/java/seedu/address/storage/JsonAdaptedModule.java

```diff
@@ -30,13 +51,27 @@ public JsonAdaptedModule(Module mod) {
      * @throws IllegalValueException If moduleCode is null.
      */
     public Module toModelType() throws IllegalValueException {
-        if (moduleCode == null) {
+        if (moduleCode == null || moduleName == null || moduleCredit == null) {
             throw new IllegalValueException(MISSING_MODULE_MESSAGE_FORMAT);
         }
         if (!ModuleCode.isValidModuleCode(moduleCode)) {
             throw new IllegalValueException(ModuleCode.MODULE_CODE_CONSTRAINTS);
         }
+        if (!ModuleName.isValidModuleName(moduleName)) {
+            throw new IllegalValueException(ModuleName.MODULE_NAME_CONSTRAINTS);
+        }
+        int integerModuleCredit;
+        try {
+            integerModuleCredit = Integer.parseInt(moduleCredit);
+        } catch (NumberFormatException nfe) {
+            throw new IllegalValueException(ModuleCredit.MODULE_CREDIT_CONSTRAINTS);
+        }
+        if (!ModuleCredit.isValidModuleCredit(integerModuleCredit)) {
+            throw new ParseException(ModuleCredit.MODULE_CREDIT_CONSTRAINTS);
+        }
```

Comment:
Will it be better to replace this part with parseModuleCredit since they are quite similar

Reply:
I think that the JsonAdaptedModule class should not be using ParserUtil class methods, even though it might seem to fit. I would think that the ParserUtil class methods should be mainly used to parse user input. The ParserUtil class methods generally throw ParseException, while IllegalValueException should only be thrown here, which in this case I will be changing the ParseException above to an IllegalValueException.